File: bugtrancking-back/src/main/java/fr/uphf/bugtrackingweb/Developpeur.java
package fr.uphf.bugtrackingweb;

import javax.persistence.*;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@JsonIdentityInfo(
        generator = ObjectIdGenerators.PropertyGenerator.class,
        property = "id"
)
public class Developpeur
{
    @Id @GeneratedValue(strategy=GenerationType.IDENTITY)
    private int id;
    private String nom;
    private String avatar;
    @OneToMany(mappedBy = "developpeur")
    //@JsonManagedReference
    private List<Bug> bugs;
    @OneToMany(mappedBy = "developpeur")
    //@JsonManagedReference
    private List<Commentaire> commentaires;

}

File: bugtrancking-front/src/app/models/developpeur.ts
export interface Developpeur
{
  id?: number;
  nom: string;
  avatar: string;
  bugs: Array<Bug>;
  commentaires: Array<Commentaire>;
}

export interface Commentaire
{
  id?: number;
  message: string;
  developpeur: Developpeur;
  dateP: Date;
  bug: Bug;
}

export interface Bug
{
  id?: number;
  description: string,
  titre: string;
  priorite: string;
  etat: string;
  dateC: Date;
  developpeur: Developpeur;
  commentaire: Array<Commentaire>;
}
